test/e2e/multicast_test.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"antrea.io/antrea/pkg/features"
+)
+
+func skipIfMulticastDisabled(tb testing.TB) {
+	skipIfFeatureDisabled(tb, features.Multicast, true, false)
+}
+
+func TestMulticast(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+	skipIfNotIPv4Cluster(t)
+	skipIfMulticastDisabled(t)
+	skipIfNumNodesLessThan(t, 2)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	t.Run("testMulticastBetweenPods", func(t *testing.T) {
+		testcases := []multicastTestcase{
+			{
+				name:                   "testMulticastForLocalPods",
+				clientPodOnNodeIdx:     0,
+				isClientPodHostNetwork: false,
+				serverPodOnNodeIdx:     0,
+				isServerPodHostNetwork: false,
+				port:                   3456,
+				group:                  net.ParseIP("224.3.4.5"),
+			},
+			{
+				name:                   "testMulticastForInterNodePods",
+				clientPodOnNodeIdx:     0,
+				isClientPodHostNetwork: false,
+				serverPodOnNodeIdx:     1,
+				isServerPodHostNetwork: false,
+				port:                   3457,
+				group:                  net.ParseIP("224.3.4.6"),
+			},
+			{
+				name:                   "testMulticastTrafficFromExternal",
+				clientPodOnNodeIdx:     0,
+				isClientPodHostNetwork: true,
+				serverPodOnNodeIdx:     1,
+				isServerPodHostNetwork: false,
+				port:                   3458,
+				group:                  net.ParseIP("224.3.4.7"),
+			},
+			{
+				name:                   "testMulticastTrafficToExternal",
+				clientPodOnNodeIdx:     0,
+				isClientPodHostNetwork: false,
+				serverPodOnNodeIdx:     1,
+				isServerPodHostNetwork: true,
+				port:                   3459,
+				group:                  net.ParseIP("224.3.4.8"),
+			},
+		}
+		for _, mc := range testcases {
+			mc := mc
+			t.Run(mc.name, func(t *testing.T) {
+				t.Parallel()
+				runTestMulticastBetweenPods(t, data, mc)
+			})
+		}
+	})
+
+	t.Run("testMulticastBetweenPodsMultipleReceivers", func(t *testing.T) {
+		testcases := []multicastMultipleReceivers{
+			{
```

Comment:
Add a case with multiple receivers on the same Node not hostNetwork?

Reply:
Added. Call it `testMulticastMultipleReceiversOnSameNode`